docs/source/howto/run_containerized_codes.rst

```diff
@@ -0,0 +1,160 @@
+
+.. important::
+
+    Running code in a container usually requires mapping the current directory (where the input files are located) to the working directory inside the container.
+    The current directory is specified in the job script as ``$PWD``.
+    The computer setup to run containerized code **MUST** set ``use_double_quotes`` to ``True`` to escape the command line parameters using double quotes.
+    The default are single quotes, but that will prevent the ``$PWD`` expression from being evaluated and instead will be treated as a string literal.
+    If the computer you want to use has already been configured, you can change the quote type in the API:
+
+    .. code-block:: python
+
+        computer = load_computer()
+        computer.set_use_double_quotes(True)
+
+
+There are two ways that a containerized code can be configured in AiiDA:
+
+* The container, which already contains the code to be run, is pre-installed on a specific target computer (``installed``)
+* The container provides an execution environment into which a particular code (executable and files) is copied before it is executed (``portable``)
+
+These two types are referred to as ``installed`` and ``portable``, because an ``installed`` container is already installed on a target computer and so it will be tied to a specific :class:`~aiida.orm.computers.Computer` configured in AiiDA, whereas the ``portable`` code is not yet installed on any computer but can be retrieved from a container index (such as Docker Hub) and downloaded and run into any computer.
+The ``portable`` code therefore does not require defining any particular :class:`~aiida.orm.computers.Computer` but can be sent to any configured computer when at calculation launch time.
+These two concepts are implemented by the :class:`aiida.orm.nodes.data.code.containerized.InstalledContainerizedCode` and :class:`aiida.orm.nodes.data.code.containerized.PortableContainerizedCode` data plugins.
+They are analogs to the plugins for non-containerized codes :ref:`InstalledCode <topics:data_types:core:code:installed>` and :ref:`PortableCode <topics:data_types:core:code:portable>`.
+
+
+.. note::
+
+   For the purposes of this how-to guide, we will be using `Singularity <https://singularity-docs.readthedocs.io/en/latest/>`__ as the containerizing technology.
+   Singularity is easy to install and designed to support scientific computing on HPC compute resources.
+   There are other similar containerizing technologies for HPC, such as `Sarus <https://sarus.readthedocs.io/en/stable/>`__ and `Shifter <https://github.com/NERSC/shifter>`__ which are equally supported by the containerized code plugins, although only Sarus and Singularity have been explicitly tested.
```

Comment:
Let's mention enroot here as well https://github.com/NVIDIA/enroot/ (if singularity works I believe it will work as well).

Plus: although docker is not targeting HPC, it is the most popular container runtime by a mile, so we need to at least mention whether it is supported or not.